src/main/java/net/neoforged/neoforge/common/data/RecipePrioritiesProvider.java

```diff
@@ -32,10 +32,10 @@ public abstract class RecipePrioritiesProvider implements DataProvider {
     private final Map<ResourceLocation, Integer> toSerialize = new LinkedHashMap<>();
     private boolean replace = false;
 
-    public RecipePrioritiesProvider(PackOutput output, CompletableFuture<HolderLookup.Provider> registries, String modid) {
+    public RecipePrioritiesProvider(PackOutput output, CompletableFuture<HolderLookup.Provider> registries) {
```

Comment:
For 1.21.4 we shouldn't make a breaking change.

Reply:
My recommendation would be to leave this parameter in, deprecate this constructor (the `@Deprecated(forRemoval = true, since = "1.21.4")` annotation plus a `@deprecated` javadoc explaining what to use instead), and provide a constructor that does not have the unused mod ID parameter.